sdk/src/main/scala/com/akkaserverless/javasdk/impl/DiscoveryImpl.scala

```diff
@@ -67,6 +80,7 @@ class DiscoveryImpl(system: ActorSystem, services: Map[String, Service]) extends
       // only (silently) send service info for hybrid proxy version probe
       Future.successful(Spec(serviceInfo = Some(serviceInfo)))
     } else {
+      proxyTerminatedRef.getAndSet(Promise[Done]()).trySuccess(Done)
```

Comment:
Not sure the `getAndSet` and `trySuccess` on the old value is needed, given that it's already set to successful promise, but no harm in doing so I guess.

Reply:
yeah, I was just thinking about if the proxy would go havoc and making multiple new discovery calls while the shutdown is in progress (waiting on old promise instance). Doesn't hurt to try to complete the old promise. Anyhow, it's not a big deal if there would be a race because in the end there is a timeout of 5 seconds for the coordinated shutdown phase.

Reply:
cool, makes sense